espnet2/spk/encoder/rawnet3_encoder.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 Jee-weon Jung
+# Apache 2.0
+
+"""RawNet3 Encoder"""
+
+import torch
+import torch.nn as nn
+from asteroid_filterbanks import Encoder, ParamSincFB
+from typeguard import check_argument_types
+
+from espnet2.asr.encoder.abs_encoder import AbsEncoder
+from espnet2.spk.layers.RawNetBasicBlock import Bottle2neck, PreEmphasis
+
+
+class RawNet3Encoder(AbsEncoder):
+    """
+    RawNet3 encoder. Extracts frame-level RawNet embeddings from raw waveform.
+    paper: J. Jung et al., "Pushing the limits of raw waveform speaker
+        recognition", in Proc. INTERSPEECH, 2022.
+
+    Note that the model's output dimensionality self._output_size equals to
+        1.5 * ndim.
+
+    Args:
+        block: type of encoder block class to use.
+        model_scale: scale value of the Res2Net architecture.
+        ndim: dimensionality of the hidden representation.
+        sinc_stride: stride size of the first sinc-conv layer where it decides
+            the compression rate (Hz).
+    """
+
+    def __init__(
+        self,
+        block=Bottle2neck,
+        model_scale: int = 8,
+        ndim: int = 1024,
```

Comment:
How about still using `output_size` and inferring `ndim` as `output_size * 2 // 3`? This looks more intuitive in the configuration.

Reply:
Thanks for the suggestion! But I think after the modifications and more docstrings, current status can be okay (and I want to avoid `//3` :)